src/app/Trade.ts

```diff
@@ -0,0 +1,70 @@
+import { AnimalNames } from '../Enums/AnimalNamesEnum';
+import { ConvertToAnimalObject } from './utils/ConvertToAnimalObject';
+import { multiply } from 'lodash';
+import { Player } from '../Player';
+import { Herd } from './logic/Herd';
+
+export class Trade {
+  constructor(private bank: Player) {}
+  /**
+   * Gets an offer from player and returns true or false if transaction can be made processed and process it if possible
+   * @param offer accepts tuple with offer containing animal name and quantity to be sold
+   * @param player accepts instance of Player class, which wants to sell his animals
+   * @param target accepts tuple with desired animal(s) containing desired animal name and quantity to buy
+   * @returns true if transaction will be processed, and false otherwise
+   */
+  processOffer(
+    offer: [AnimalNames, number],
+    { theHerd: playerHerd }: Player,
+    target: [AnimalNames, number],
```

Comment:
Great idea with destructuring first and third argument in disposeResult method, maybe the same could be applied here.